behaviors/services/crosschain-connector.md

```diff
@@ -42,3 +42,47 @@ Runs nodes for other blockchains like Ethereum and provides read access to them.
   * Emitting contract address
   * Event name
 * Returns a list of events from the desried transaction's receipt that matches the event signature and emitting contract.
+
+
+
+
+&nbsp;
+## `EthereumGetBlockInfoByTime` (method)
+> Retrieve info about the latest Ethereum block such that its timestamp < provided ethereum_timestamp.
+
+* Query the Ethereum node to find the appropriate block using the given arguments.
+* Comply to the finality time shift.
+    * If the provided ethereum_timestamp > current time - ETHEREUM_FINALITY_TIME_COMPONENT, fail the call.
+        * current time is provided as reference_time.
+* Returns the block info (uint64 block_number, uint64 block_timestamp) of the resulting block.
+
+
+
+&nbsp;
+## `EthereumGetLatestBlockInfo` (method)
+> Retrieve info about the latest Ethereum block such that its timestamp < reference_time under finality time shift.
+
+* Query the Ethereum node to find the appropriate block using the given arguments.
+* Comply to the finality time shift:
+    * Find the maximal Ethereum block whose timestamp < reference_time - ETHEREUM_FINALITY_TIME_COMPONENT.
+* Returns the block info (uint64 block_number, uint64 block_timestamp) of the resulting block.
+
+
+
+&nbsp;
+## `EthereumGetLogs` (method)
+> Returns all Ethereum logs entries which comply to the given filter params. 
+* Query the Ethereum node to find the appropriate logs using the given arguments.
+* Comply to the finality time shift:
+    * Get Latest finalized block by calling EthereumGetLatestBlockInfo
+    * Verify the block range provided is in the finality range.
+        * finalized block number >  until_block_number (provided in arguments).
+* Derive the event signature key from the provided contract ABI using the event_name.
+* The ABI needs to be defined for the ethereum connector to be able to make the call
+  * The ABI must contain only one event with the provided event name.
+* Returns:
+    * call_result - indicates the call success or failure.
+    * block_number - indicates the last filtered block_number.
```

Comment:
block_number - indicates the last filtered block_number. The output block number is less or equal to the requested until_block